deploy/role.yaml

```
@@ -99,6 +101,19 @@ rules:
       - list
       - watch
 
+  # Permissions needed to be able to grant to prometheus-k8s SA from openshift-monitoring NS
+  - apiGroups:
+      - ""
```

```suggestion
      - rhmi-prometheus-k8s
```
I was seeing the same error when trying to upgrade form one version to another. It got stuck reconciling on monitoring. Applied this change but had to make the above suggestion to make it work.

---

This doesn't make much sense to me.
There is no APIGroup "rhmi-prometheus-k8s".

You can read more about these groups here: https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-groups

---

As discussed in chat, this would be incorrect as the apiGroup, but the act of modifying the role on the cluster may have caused something else to reconcile and fix your issue.